### DISCUSSION TITLE
cannot import name 'deprecated' from 'typing_extensions'

### DISCUSSION THREAD

**Discussion by wocanbe** [fastapi/fastapi#9808] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

When I use fastapi in Python 3.10.11. Throw some errors like "cannot import name 'deprecated' from 'typing_extensions“

**Comment by ahmedabdou14** [comment]
Please provide more details on your error. It seems to have nothing to do with fastapi

**Reply by keul** [reply]
I've the same issue using a conda based environment. Env was working until few minutes ago, when IIrebuild it from scratch:

```
│ Traceback (most recent call last):                                                                                                                                                                       │
│     import fastapi                                                                                                                                                                                       │
│   File "/opt/conda/lib/python3.10/site-packages/fastapi/__init__.py", line 7, in <module>                                                                                                                │
│     from .applications import FastAPI as FastAPI                                                                                                                                                         │
│   File "/opt/conda/lib/python3.10/site-packages/fastapi/applications.py", line 16, in <module>                                                                                                           │
│     from fastapi import routing                                                                                                                                                                          │
│   File "/opt/conda/lib/python3.10/site-packages/fastapi/routing.py", line 22, in <module>                                                                                                                │
│     from fastapi import params                                                                                                                                                                           │
│   File "/opt/conda/lib/python3.10/site-packages/fastapi/params.py", line 6, in <module>                                                                                                                  │
│     from typing_extensions import Annotated, deprecated                                                                                                                                                  │
│ ImportError: cannot import name 'deprecated' from 'typing_extensions' (/opt/conda/lib/python3.10/site-packages/typing_extensions.py)
```

I suspect the issue can be related to this recent [changes in typing_extensions](https://github.com/python/typing_extensions/blob/main/CHANGELOG.md#release-470rc1-june-21-2023):

> typing_extensions now re-exports all names in the standard library's typing module, except the deprecated ByteString. Patch.

**Reply by Kludex** [reply]
What's the `typing_extensions` version?

**Reply by keul** [reply]
I'm trying to check, but k8s -> docker -> conda makes everything sooooo easy to be debugged 😄 

```
fastapi                   0.99.0             pyhd8ed1ab_0    conda-forge
…
typing-extensions         4.4.0                    pypi_0    pypi
typing_extensions         4.7.1              pyha770c72_0    conda-forge
```

Don't fully know why I see two typing-extensions

**Reply by keul** [reply, marked as answer]
I was wrong.

Tried this specific import only (`from typing_extensions import Annotated, deprecated`) in dedicated virtualenv.

- with typing_extensions-4.7.1 it works
- with typing_extensions-4.4.0 it raises the error

So the issue seems on my env, someway the pip installed dependency is winning over conda. Now I've to understand which is installing it.

Sorry for the noise

**Comment by HoagieFestDS** [comment]
I'm using typing_extensions 4.7.1 and am getting the same error for anyone else who might have having same issue.  Using Python 3.9, but the documentation says it's compatible

**Reply by Abecid** [reply]
I am getting this error with 4.7.1 (Python 3.8)

**Reply by andreped** [reply]
Just got the same error in a static project. Anyone found a solution?

**Reply by andreped** [reply]
Upgrading to `typing_extensions==4.7.1` for `Python==3.7` resolved the issue. I force-upgraded `typing_extensions` after installing all other dependencies.
Strange that this broke, when I kept all my versioning strict...

**Reply by HoagieFestDS** [reply]
I'm using 3.9.  Haven't tried 3.7, but seems odd (or at least annoying) that >= 3.8 would have issues with this

**Reply by keul** [reply]
As you can see above, I had the same issue with Python 3.10, but the problem was conda. So, if you are using conda I suggest you to look in this direction.

I was not able to understand which dependency was installing a second version of typing_extensions using pip (because conda is 🤷‍♂️), but after upgrading some conda packages, the issue disappeared.

**Reply by HoagieFestDS** [reply]
Thanks.  I'm not using Conda, but will try to take a look at other dependencies are also installing typing_extensions

**Reply by mattielang** [reply]
Deleting and reinstalling typing_extensions==4.7.1 with pip fixed it for me with Python 3.10.  

Was originally installing via conda

**Comment by 907Resident** [comment]
> Deleting and reinstalling typing_extensions==4.7.1 with pip fixed it for me with Python 3.10.
> 
> Was originally installing via conda

Nice! This was it. Even though I already had `typing_extensions==4.7.1` installed but was gettin the error mentioned at the top of the thread. 

However, running `pip uninstall typing_extensions --yes` and the reinstalling `typing_extensions==4.7.1` did the trick for me.

**Reply by Vvsmile** [reply]
why?? I met the same problem after I used ''conda create --clone'', but I checked the new cloned directory, there is the typing_extensions..., what a stranger bug it is!

**Comment by wocanbe** [comment]
typing_extensions version issues, It can be resolved through commands. `pip install typing_extensions==4.7.1 --upgrade`

**Reply by mabreuortega** [reply]
For me it is happening in databricks cluster, there is a /databricks/python/lib/python3.10/site-packages/typing_extensions.py which is trying to load deprecated from instead of running the installed package

**Reply by HoagieFestDS** [reply]
I was doing this through Databricks as well.  I was able to get it to work on Runtime 10.4, which unfortunately is a deprecated runtime.  But it still worked.

**Reply by kneuskens** [reply]
and did you solve it on the databricks cluster? i am running on python 3.9 in runtime 11.3 and tried a lot of versions of typing extensions and pydantic. always unable to import 'deprecated' or 'required'

**Reply by HoagieFestDS** [reply]
Yes - I had to use Runtime 10.4

**Reply by kneuskens** [reply]
indeed that does the trick. thanks

**Reply by mcnewcp** [reply]
I was also having this issue on a databricks cluster and I resolved it with:
```
! pip install typing-extensions==4.7.1
dbutils.library.restartPython()
```

**Comment by mabreuortega** [comment]
In my case I'm using pydantic and was able to work around the issue with pydantic<2.3.0
Which includes the deprecated import

**Comment by Hopper1906** [comment]
I would LOVE to use **ydata-profiling**'s Profile Report

On Google Colab today running Python 3.10.12, 
On:
**from ydata_profiling import ProfileReport**

I see the error: 
_ImportError: cannot import name 'TypeAliasType' from 'typing_extensions' (/usr/local/lib/python3.10/dist-packages/typing_extensions.py)_


Related info: 

**Environment**
On Google Colab today running Python 3.10.12

Importa
import numpy as np
import  pandas as pd

#Begin installs to try to reproduce the slick data Profile report from the FitBit code 
!pip install fastapi
!pip install kaleido
!pip install uvicorn
!pip install python-multipart
!pip install ydata-profiling
!pip install cohere
!pip install openai
!pip install tiktoken
#!pip install typing-extensions
!pip install typing-extensions --upgrade
from typing_extensions import Annotated, deprecated
from ydata_profiling import ProfileReport
#End installs ... Try to reproduce the slick data Profile report from the FitBit code

**Reply by a3131919** [reply]
I got the same error until install 4.8.0

typing_extensions         4.8.0
Python 3.11.5 

fixed it.

**Reply by wanghaisheng** [reply]
@a3131919 what is fastapi version

**Comment by NextoneX** [comment]
In my case, I meet the same question: cannot import name 'deprecated' from 'typing_extensions'.
I'm using conda with python environment 3.9.12. But I cannot fix this problem by using `pip install typing-extensions==4.7.1`
and also fastapi 0.105.0 requires typing-extensions>=4.8.0 now.

**Reply by wanghaisheng** [reply]
try python 3.10 >

**Comment by johnyzv** [comment]
In my case bumped into this error after upgraded to pydantic 2 and happened only on aws lambda runtime, local env and unit tests worked just fine.
Issue was caused by the lambda layers, more specifically I had layer1 (python requirements) and typing_extensions==4.7.1 and layer2 (opentelemetry) with typing_extensions==4.3.0 in this exact order. This caused one layer to override another layer `typing_extensions` while they still reported version 4.7.1 for some reason. The fix was to invert the order precedence of the layers, basically layer2 first and layer1 second.

**Comment by RubTalha** [comment]
https://stackoverflow.com/a/77075595/13086128
